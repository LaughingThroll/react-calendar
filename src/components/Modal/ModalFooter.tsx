import React from "react"
import plus from "../../assets/images/icons/plus.svg"

import { Button } from "../common"

const ModalFooter = () => {
  return (
    <div className="modal-footer">
      <Button secondary>Cancel</Button>
      <Button type="submit">Send</Button>
    </div>
  )
}

export default ModalFooter
