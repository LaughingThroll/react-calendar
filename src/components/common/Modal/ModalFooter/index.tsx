import React from 'react'
import styles from './modalFooter.module.scss'

interface ModalFooterProps {
  children?: JSX.Element | JSX.Element[]
  className?: string[]
}

const { modalFooter } = styles

const ModalFooter: React.FC<ModalFooterProps> = ({ children, className = [] }) => (
  <div className={`${modalFooter} ${className.join(' ')}`}>{children}</div>
)

export default React.memo(ModalFooter)
