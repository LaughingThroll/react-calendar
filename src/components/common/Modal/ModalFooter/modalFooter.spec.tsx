import React from 'react'
import { shallow } from 'enzyme'
import ModalFooter from './'

describe('ModalFooter render', () => {
  it('render default', () => {
    const tree = shallow(<ModalFooter />)

    expect(tree).toMatchSnapshot()
    expect(tree.find('div')).toHaveLength(1)
  })

  it('render with children', () => {
    const tree = shallow(<ModalFooter children={<span>Test</span>} />)

    expect(tree).toMatchSnapshot()
    expect(tree.find('span')).toHaveLength(1)
  })
})

describe('ModalFooter props', () => {
  it('component wiht className', () => {
    const tree = shallow(<ModalFooter className={['hello']} />)

    expect(tree.find('div').hasClass('hello')).toBe(true)
  })
})
