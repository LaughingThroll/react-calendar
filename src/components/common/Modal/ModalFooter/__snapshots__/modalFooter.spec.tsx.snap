// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalFooter render render default 1`] = `
<div
  className="modalFooter "
/>
`;

exports[`ModalFooter render render with children 1`] = `
<div
  className="modalFooter "
>
  <span>
    Test
  </span>
</div>
`;
